internal/pkg/service/appproxy/appconfig/api.go

```diff
@@ -0,0 +1,122 @@
+package appconfig
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/keboola/go-client/pkg/request"
+	"github.com/pquerna/cachecontrol/cacheobject"
+)
+
+type AppProxyConfig struct {
+	ID              string         `json:"id"`
+	Name            string         `json:"name"`
+	UpstreamAppHost string         `json:"upstreamAppHost"`
+	AuthProviders   []AuthProvider `json:"authProviders"`
+	AuthRules       []AuthRule     `json:"authRules"`
+	eTag            string
+	maxAge          time.Duration
+}
+
+type AuthProvider struct {
+	ID           string   `json:"id"`
+	Name         string   `json:"name"`
+	Type         string   `json:"type"`
+	ClientID     string   `json:"clientId"`
+	ClientSecret string   `json:"clientSecret"`
+	IssuerURL    string   `json:"issuerUrl"`
+	AllowedRoles []string `json:"allowedRoles"`
+}
+
+type AuthRule struct {
+	Type  string   `json:"type"`
+	Value string   `json:"value"`
+	Auth  []string `json:"auth"`
+}
+
+func GetAppProxyConfig(sender request.Sender, appID string, eTag string) request.APIRequest[*AppProxyConfig] {
```

Comment:
Is the pointer `*AppProxyConfig` needed here? We prefer to pass "entity" as value if possible.

Reply:
Ok, it may be that it is not solved perfectly there.